doc/new-guidelines.md

```diff
@@ -784,6 +784,24 @@ src
         └── view.cljs
 ```
 
+### Deprecation process
+
+To properly indicate that a function is deprecated, add the ^:deprecated metadata to it. This not only formally marks the function as deprecated but also enables us to utilize linting tools to prevent its use in new code. One of the key benefits of this approach is the ability to include inline comments guiding developers to the recommended alternative function.
```

Comment:
The original intent of the guidelines was to be a quick and easy document to read/write, otherwise there's an ever increasing chance of it being outdated and/or nobody reading it. We'are basically following what's in the Clojure Style Guide for section https://guide.clojure.style/#deprecated, so we don't need to be verbose.

If we want to have an extended explanation, we use the `details` HTML tag to avoid polluting the docs with too much detail.

```
## Deprecation process

To deprecate a var, add the `:deprecated` metadata and, if necessary, suggest an
alternative.

```clojure

;; Good if there's no better alternative yet, but we want to deprecate it anyway.
(defn ^:deprecated foo
  []
  (bar))

;; Good
(defn foo
  {:deprecated "Use some.namespace/var-name instead."}
  []
  (bar))

<details>
  <summary>Expand to see why.</summary>
  bla bla bla
</details>
```